src/ptracer.rs

```diff
@@ -397,22 +397,21 @@ impl Ptracer {
                         let ret_data = ptrace::getevent(pid)? as u16;
                         let stop = Stop::Seccomp(ret_data);
 
-                        match self.tracee_state_mut(pid) {
-                            Some(state) if *state == State::Attaching => {
-                                *state = State::Syscalling;
-                            },
-                            _ => internal_error!(),
+                        if let Some(state) = self.tracee_state_mut(pid) {
+                            *state = State::Syscalling;
```

Comment:
We should arguably be special-casing `Attaching` and `Spawning`, but this is less wrong than the status quo, which would bail with an internal error if the tracee was in the `Traced` state(!)